src/curves.rs

```diff
@@ -50,6 +56,8 @@ macro_rules! new_curve_impl {
         /// infinity).
         #[derive(Copy, Clone)]
         #[cfg_attr(feature = "repr-c", repr(C))]
+        #[cfg_attr(feature = "borsh", derive(BorshSerialize, BorshDeserialize))]
+        #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
```

Comment:
This is unsafe, as it allows callers to deserialize affine points that are not on the curve. We should instead have manual implementations of these, which go through `group::GroupEncoding::from_bytes` and maps the `None` case of `CtOption` to a deserialization error.